### ISSUE TITLE
CA1859 shouldn't be triggered by anonymous types

### ISSUE THREAD

**Issue by alexmurari** [dotnet/roslyn-analyzers#7504]
### Analyzer

**Diagnostic ID**: [CA1859](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1859): `Use concrete types when possible for improved performance`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 8.0.0

### Describe the bug

When a method's return type is `object` and the returned value is an anonymous type (`new { Name = "Alex" }`) the analyzer suggests changing the method's return type from `object` to `<anonymous type: string Name>`. Of course, this return type is invalid.

### Steps To Reproduce

```csharp
Console.WriteLine(BuildObject());

private static object BuildObject() // Triggers CA1859
{
    var outputData = new
    {
        Name = "Alex"
    };

    return outputData;
}
```

### Expected behavior

Shouldn't trigger the CA1859 rule.

## Additional context

Triggers the CA1859 rule.
